### DISCUSSION TITLE
[Feature] Age of a person at date of photo

### DISCUSSION THREAD

**Discussion by daniele-athome** [immich-app/immich#3348] [Feature Request]
I'd like to start a discussion about a feature that I'd would like to see in immich - and I might be even able to contribute, but first I want to collect some feedback by the community and the core developers.

### The feature

It would be nice to show the age of a person in a photo at the time that photo was taken. The date of birth could be added manually as ~a tag (I understand that manual tagging via UI is still not possible though, but there is an API so one could experiment... also tags maybe weren't designed for this kind of use?).~ an attribute of the person.

### Platform

- [X] Server
- [X] Web
- [X] Mobile

**Comment by jrasm91** [comment]
It seems easy enough to make it possible to store a birthday for a person. Not 100% sure what you would do with it though. Do you have some examples of where you would use it in the app and how it would look?

**Reply by daniele-athome** [reply]
What I'd like to see is the age of a person in a photo at the time the photo was taken, so I was thinking about the photo info panel.

For the web app, I'd imagine showing the age (and I mean something like "34 years old", "age(d) 34", or something similar) just below the person photo (or the name, if available) in the info panel (very quick and ugly mockup via browser hacking):

![example-age](https://github.com/immich-app/immich/assets/3763726/3263f05d-f21f-4ddc-99e8-5218f00dc585)

For the mobile app, since people are not displayed in the info panel, I'm finding it hard to find a place for it.

**Reply by martabal** [reply]
I feel this is out of Immich scope. Does google photo have this feature ?

I don't intend for Immich to serve as my primary address book, but rather to integrate with my address book through protocols like CardDAV.

**Reply by alextran1502** [reply]
I have the same feeling with @martabal, and I am afraid trying to include this information would cluster the interface, introduce more information to manage and maintain on the backend.

**Reply by jrasm91** [reply]
I don't mind this feature myself, but it does start to blur the line a bit between photo management software and contacts.

**Reply by daniele-athome** [reply]
Thank you all for your feedback.

> Does google photo have this feature?

I don't know, but if it doesn't they should implement it. I mean one of the things that Google sells the most is the integration among their services, so for Google this might make sense.

> I don't mind this feature myself, but it does start to blur the line a bit between photo management software and contacts.

I agree. This is one of the feature that IMHO is **ON** that line: where would you show that information? In the photos app or in the contacts app (next to... what?).
Anyway, the minute you add face recognition and a table called "person" in a photos app, you start to walk towards that line.

> I don't intend for Immich to serve as my primary address book, but rather to integrate with my address book through protocols like CardDAV.

Me too, but that's a massive development to make for Immich.
After implementing this proposal, I also wanted to write a little sync tool that will connect to my CardDAV server and sync the birth date for names that match (I'll open source it of course).

Eventually, I guess some CardDAV integration could be implemented for Immich, but that's a big step. This proposal is a small (first) step.

Do you have some suggestions on how to address this so it can be accepted into Immich? If you think it's totally out of scope I'll maintain the fork, no problem.

**Reply by mPyKen** [reply]
i wonder if there is a face recognition model (or any other way) that can take age into account, e.g. two kids looking similar but having an age gap of 30y cannot be the same person. if there is sth like that, I can imagine the proposed feature being quite useful

**Comment by daniele-athome** [comment]
FWIW I started working on it in my fork. I think I will proceed in steps:

* a first PR that adds the capability of setting a birth date for a face (https://github.com/daniele-athome/immich/tree/feat/person-birth-date)
* a second PR that shows the age of the face when viewing an asset

Since I also own a CardDAV server and I don't want to input all birth dates manually, I'll also write a simple tool to sync birth dates from a CardDAV server to Immich people using a simple name match.

**Comment by daniele-athome** [comment]
#3721 was just merged containing both things: setting a birthdate for a face and showing the age of the person.

I'll be working on the CardDAV sync tool here: https://github.com/daniele-athome/immich-carddav-sync

I'm closing this is as resolved since it's been implemented in Immich.

**Comment by daniele-athome** [comment]
Hi! A first **alpha** version of the tool is now available. Docker usage guide here: https://github.com/daniele-athome/immich-carddav-sync#usage-with-docker

**Comment by codetalker78** [comment]
I was researching exactly this use case: I have a lot of photos of my kids from over the years and it would be really nice to see their age in each picture.
Excellent that immich supports this!

Besides displaying the persons age in the UI, the birth date could also be used for person detection as a decision criteria - persons cannot be on a photo taken before their date of birth! So the algorithm could discard such a potential face match and select the next best candidate! 
As a new user, I have imported photos from the baby time of all my kids and got a lot of mismatches between them. A sanity check like this "is this kid (person) already alive at the time the picture was taken" could have prevented most of the misclassifications. 
I am new to Immich. Would this be a something for which a feature request makes sense?

**Reply by bo0tzz** [reply]
An FR for that already exists.

**Reply by broecher** [reply]
It exist in immich or just it exists somewhere?

**Comment by Gustry** [comment]
I just want to say thanks for this feature ! 

I have a lot of old pictures to set an very approximated date, it helps a lot !

I'm doing a kind of "computation" in my mind to set a date according to the age of someone on the picture (I might create a ticket for this feature request on this, to set a new date based on the age of someone), but it the meantime, it's a good feature.
